### ISSUE TITLE
3.22.0: Mutable overlays disappear from map after editing

### ISSUE THREAD

**Issue by Moonbase59** [valentine195/obsidian-leaflet-plugin#190]
After creating a mutable overlay, right-clicking to edit it, and closing the edit modal … it’s vanished. (Somehow reminds me of an old bug?)

But it stores them correctly in data.json, and when closing/reopening the map they’re there. But always vanish after editing.

**Comment by valentine195** [comment]
Please try 3.22.2

**Comment by Moonbase59** [comment]
![Auswahl_166](https://user-images.githubusercontent.com/3706922/128075238-15a7fddd-5a9e-4339-9ce5-d06df03998ba.png)

Now it offers "Overlay 1" when clicking on "Overlay 2", and after editing (it edits "Overlay 2"), it’s gone again.

**Comment by valentine195** [comment]
The overlays in the context menu are just given generic `overlay <index>` names, they don't actually correspond to anything.

**Comment by Moonbase59** [comment]
Ah, ok. Thought they’d pick up the overlay names. Apparently we had the same idea in naming them. Also I now understand the visual aid to show the overlay when hovering on the context menu.

Anyway, they still disappear.

**Comment by valentine195** [comment]
Are they still in the DOM or completely removed?

**Comment by Moonbase59** [comment]
They’re still in the DOM.

**Comment by valentine195** [comment]
> M-278.5,352.90624984604074a1,1 0 1,0 2,0 a1,1 0 1,0 -2,0

Looks like it loses its radius. Good spot.

`M`: Move here in absolute coordinates.
`a`: Draw an elliptical arc using relative coordinates.

`a1,1` says draw an elliptical arc with a radius equal to 1 pixel from my current position.

**Comment by Moonbase59** [comment]
Good spot back to you!

**Comment by valentine195** [comment]
Can you try editing an overlay in 3.23.1 in verbose mode? Added some logging.

**Comment by Moonbase59** [comment]
Yup. Here we go:

Right-clicked and edited "Overlay 2" (left to the visible one), immediately closed edit modal. "Overlay 2" gone but still in the DOM (apparently 2x2 px).

![1_088](https://user-images.githubusercontent.com/3706922/128355779-9add1ab6-d3a7-460d-bd33-131527d5b46d.png)

Console:

```text
app.js:1 Obsidian Developer Console
main.ts:27 loading markdown-table-editor plugin
main.ts:40 Dataview: Version 0.4.x Loaded
main.ts:67 DiceRoller plugin loaded
main.ts:49 loading Find unlinked files and unresolved links plugin
main.ts:40 loading Hotkeys for templates plugin v1.4.2
main.ts:10 Loading Hotkeys++ plugin
main.ts:125 loading plugin
VM199:2 Loading Obsidian Leaflet v3.23.1
main.ts:55 loading plugin
main.ts:17 Sortable: loading plugin...
main.ts:25 Sortable: loaded plugin.
main.ts:17 Loading snippets plugin
main.ts:27 Loading Pandoc plugin
main.ts:28 Loading status bar pomodoro timer
VM230:4563 loading dictionary
main.ts:9 loading ActiveNoteTitlePlugin plugin
DevTools failed to load SourceMap: Could not parse content for app://obsidian.md/main.js.map: Unexpected end of JSON input
5[Violation] Added non-passive event listener to a scroll-blocking <some> event. Consider marking event handler as 'passive' to make the page more responsive. See <URL>
VM231:14294 [Longform] Loaded and watching projects.
index.ts:356 Dataview: Parsed all file prefixes (0.001s)
main.ts:148 Hotkeys for templates -> active plugins: templater-obsidian,core
index.ts:409 Dataview: Initialized CSV row cache (loading 5 CSV files).
index.ts:247 Dataview: Task & metadata parsing queued in 0.006s.
[Violation] 'requestAnimationFrame' handler took 63ms
9[Violation] Forced reflow while executing JavaScript took <N>ms
main.ts:43 tags ["map"]
main.ts:43 position {start: {…}, end: {…}}
main.ts:52 {filename: "-Map.md", filepath: "-Map.md", vault: "Knowledgebase", workspace: "", frontmatter.tags: Array(1), …}filename: "-Map.md"filepath: "-Map.md"frontmatter.position: {start: {…}, end: {…}}end: {line: 2, col: 3, offset: 19}start: {line: 0, col: 0, offset: 0}__proto__: Objectfrontmatter.tags: ["map"]0: "map"length: 1__proto__: Array(0)vault: "Knowledgebase"workspace: ""__proto__: Object
cm-addons.js:6577 [Violation] 'setTimeout' handler took 95ms
VM199:2 Obsidian Leaflet Map Main Map: Beginning Markdown Postprocessor.
VM199:2 Obsidian Leaflet Map Main Map: Loading GeoJSON files.
[Violation] 'requestAnimationFrame' handler took 161ms
VM199:2 Obsidian Leaflet Map Main Map: Building map instance.
VM199:2 Obsidian Leaflet Map Main Map: Loading immutable items.
VM199:2 Obsidian Leaflet Map Main Map: Found 60 markers and 71 overlays from 59 files.
VM199:2 Obsidian Leaflet Map Main Map: Getting initiatial coordinates.
VM199:2 Obsidian Leaflet Map Main Map: Coordinates file found.
VM199:2 Obsidian Leaflet Map Main Map: Beginning render process.
VM199:2 Obsidian Leaflet Map Main Map: Building initial map layer.
VM199:2 Obsidian Leaflet Map Main Map: MarkdownRenderChild loaded. Appending map.
VM199:2 Obsidian Leaflet Map Main Map: Map element is off the page and not loaded into DOM. Will auto-detect and reset zoom.
VM199:2 Obsidian Leaflet Map Main Map: Moving to supplied coordinates: 48.2418,10.3632
VM199:2 Obsidian Leaflet Map Main Map: Adding 61 markers to map.
VM199:2 Obsidian Leaflet Map Main Map: 61 markers added to map.
VM199:2 Obsidian Leaflet Map Main Map: Adding 73 overlays to map.
VM199:2 Obsidian Leaflet Map Main Map: 73 overlays added to map.
VM199:2 Obsidian Leaflet Map Main Map: Sorting overlays.
VM199:2 Obsidian Leaflet Map Main Map: Overlays sorted.
VM199:2 Obsidian Leaflet Map Main Map: Adding 7 GeoJSON features to map.
VM199:2 Obsidian Leaflet Map Main Map: 7 GeoJSON features added to map.
VM199:2 Obsidian Leaflet Map Main Map: Initial map layer rendered in 0.265 seconds.
[Violation] 'requestAnimationFrame' handler took 65ms
[Violation] 'requestAnimationFrame' handler took 87ms
VM199:2 Obsidian Leaflet Map Main Map: Updating the overlay data.
VM199:2 Obsidian Leaflet Map Main Map: Overlay color set to: rgba(218, 165, 32, .6)
VM199:2 Obsidian Leaflet Map Main Map: Overlay radius set to: 13.352945595074663
VM199:2 Obsidian Leaflet Map Main Map: Overlay description set to: Overlay 2
VM199:2 Obsidian Leaflet Map Main Map: Overlay tooltip set to: null
```

**Comment by valentine195** [comment]
does that radius look correct?

**Comment by Moonbase59** [comment]
The `13.352945595074663` (km) yes, the 2x2px not ;-) Should result in roughly 200x200px at this zoom.

**Comment by valentine195** [comment]
Yeah, my guess is this could be a bug that has been present for quite a long time and is unrelated to the 3.22 update. That `13.35...` number should be the number in _meters_. Could you share your map block parameters?

**Comment by Moonbase59** [comment]
Sure:

```leaflet
id: Main Map
markerFolder: People
markerFolder: Locations
markerFolder: Events
#markerFolder: -Inbox
height: 500px
lat: 48.23875
long: 10.37595
minZoom: 1
maxZoom: 18
defaultZoom: 12
unit: km
distanceMultiplier: 1.3
overlayTag: nearby
overlayColor: 'rgba(218, 165, 32, .6)'
coordinates: [[-Current Location]]
zoomTag: nearby
zoomDelta: 1
geojson:
  #- [[geo-bayern.geojson]]
  - [[geo-bayern-landkreise.geojson]]
  #- [[geo-lk-günzburg.geojson]]
  - [[Kneipp-Rundweg Krumbach.geojson]]
  - [[Lehrpfad Krumbach.geojson]]
  - [[Walderlebnispfad Krumbach.geojson]]
  - [[Gewässer-Informations-Erlebnis-Pfad Münsterhausen.geojson]]
  - [[Krumbach Oberes Gemeindeholz & Weiher.geojson]]
  - [[Krumbach Oberes Gemeindeholz.geojson]]
  - [[Kleine Runde Kellerweg.geojson]]
#linksTo: [[note]] #markers from all notes linking TO this note
#linksFrom: [[note]] #markers from all notes linking FROM this note
verbose: true
```

**Comment by valentine195** [comment]
lol no need! got it to reproduce - it was as I thought, an old bug that hadn't been caught. I wasn't seeing it because the overlay has to be drawn on a real-world map with a `unit` set.

**Comment by Moonbase59** [comment]
Can confirm that: When I change the `13.4` to `13.41` in the edit it _stays_. Really that ole bug again … lol

**Comment by valentine195** [comment]
Yeah..... I'm in the process of doing a big rewrite actually, the code has become rather unwieldy. Will make it much easier to maintain and find these kinds of bugs.

**Comment by valentine195** [comment]
Should be fixed in 3.23.2
